lib/matplotlib/colorizer.py

```diff
@@ -255,25 +248,28 @@ def set_clim(self, vmin=None, vmax=None):
         vmin, vmax : float
              The limits.
 
-             The limits may also be passed as a tuple (*vmin*, *vmax*) as a
-             single positional argument.
+             For scalar data, the limits may also be passed as a
+             tuple (*vmin*, *vmax*) single positional argument.
 
              .. ACCEPTS: (vmin: float, vmax: float)
         """
+        if self.norm.n_components == 1:
+            if vmax is None:
+                try:
+                    vmin, vmax = vmin
+                except (TypeError, ValueError):
+                    pass
         # If the norm's limits are updated self.changed() will be called
         # through the callbacks attached to the norm, this causes an inconsistent
         # state, to prevent this blocked context manager is used
```

Comment:
Not sure why this moved, but I think you can delete this comment since it says the same thing over the _actual_ `blocked` context manager.